app/controllers/api/v1/products_controller.rb

```diff
@@ -48,7 +56,7 @@ def set_product
 
       # Only allow a list of trusted parameters through.
       def product_params
-        params.require(:product).permit(:name)
+        params.require(:data).permit(:name, product_details: [])
```

Comment:
Ticket TS-708
The endpoint should receive an array of product details.

Probably not the right spot to put my comment, but the endpoint is receiving an array of product details. The test (products_controller_spec.rb) would not be successful if the endpoint would not receive an array of product details.

In general we could question the use of the array, as we always only want to have exactly one product detail for a product. (according to the Ticket)